aiida/common/timezone.py

```diff
@@ -8,132 +8,76 @@
     """
-    if timezone is None:
-        timezone = get_current_timezone()
-    if hasattr(timezone, 'localize'):
-        return timezone.localize(value, is_dst=is_dst)
-
-    if is_aware(value):
-        raise ValueError(f'make_aware expects a naive datetime, got {value}')
+    return make_aware(value)
 
-    # This may be wrong around DST changes!
-    # See http://pytz.sourceforge.net/#localized-times-and-date-arithmetic
-    return timezone.localize(value)
 
+def timezone_from_name(name: str) -> tzinfo:
+    """Return a :class:`datetime.tzinfo` instance corresponding to the given timezone name.
 
-def localtime(value, timezone=None):
-    """Converts an aware datetime.datetime to local time.
-
-    Local time is defined by the current time zone, unless another time zone is specified.
+    :param name: The timezone name. Should correspond to a known name in the Olsen database.
+        https://en.wikipedia.org/wiki/Tz_database
+    :returns: The corresponding timezone object.
+    :raises ValueError: if the timezone name is unknown.
     """
-    if timezone is None:
-        timezone = get_current_timezone()
-
-    # If `value` is naive, astimezone() will raise a ValueError, so we don't need to perform a redundant check.
-    value = value.astimezone(timezone)
-    if hasattr(timezone, 'normalize'):
-        # This method is available for pytz time zones.
-        value = timezone.normalize(value)
-
-    return value
+    try:
+        return pytz.timezone(name)
+    except pytz.exceptions.UnknownTimeZoneError as exception:
+        raise ValueError(f'unknown timezone: {name}') from exception
 
 
-def delta(from_time, to_time=None):
+def delta(from_time: datetime, to_time: datetime = None) -> timedelta:
     """Return the datetime object representing the different between two datetime objects.
 
-    :param from_time: starting datetime object
-    :param to_time: end datetime object
-    :return: the delta datetime object
+    :param from_time: The starting datetime object.
+    :param to_time: The end datetime object. If not specified :func:`aiida.common.timezone.now` is used.
+    :return: The delta datetime object.
     """
-    if to_time is None:
-        to_time = now()
-
-    try:
-        from_time_aware = make_aware(from_time)
-    except ValueError:
-        from_time_aware = from_time
-    try:
-        to_time_aware = make_aware(to_time)
-    except ValueError:
-        to_time_aware = to_time
-
-    return to_time_aware - from_time_aware
+    return make_aware(from_time) - make_aware(to_time or now())
 
 
-def datetime_to_isoformat(value):
+def datetime_to_isoformat(value: datetime) -> str:
     """Convert a datetime object to string representations in ISO format.
 
-    :param value: a datetime object
+    :param value: A datetime object
     """
-    if value is None:
-        return None
-    return value.isoformat()
+    return value.isoformat() if value is not None else value
```

Comment:
I know that this worked like this before, but since we are changing the behavior of the other methods I think it is a good moment to ask: is this really a good idea?

I mean, since the only purpose of this method is to operate on `value`, do we really want to accept and propagate `None` values? Is there a reason why this could be useful instead of just making errors slightly more difficult to catch? Is there a specific reason why is useful to ignore the `None`?

Reply:
A second question (that perhaps renders the previous one unnecessary) is: are we just keeping these methods for backwards compatibility? I see that later when we need to convert a isoformatted string to `datetime` we use `datetime.fromisoformat` directly instead of `isoformat_to_datetime`.

If we are not using these methods as a way to wrap the `datetime` tools to have a more modular dependency on them, then maybe we should deprecate these and take them out on the next iteration?

Reply:
Yeah, I think I agree. Especially since this is simply a method on the `datetime` object itself, no point to have a warpper function. Will remove this. Idem for `isoformat_to_datetime`.